pkg/resourceprovider/provider.go

```diff
@@ -159,7 +160,14 @@ func (p *resourceProvider) GetPodMetrics(pods ...*metav1.PartialObjectMetadata)
 	resultsByNs := make(map[string][]nsQueryResults, len(podsByNsBatched))
 	for result := range resChan {
 		if result.err != nil {
-			klog.Errorf("unable to fetch metrics for pods in namespace %q, skipping: %v", result.namespace, result.err)
+			if promErr, ok := result.err.(*client.Error); ok {
```

Comment:
What's the difference between asserting the type as `client.Error` vs `prom.Error`?

Reply:
There isn't a prom.Error, but the client.Error is the custom Error type defined in types.go
the custom error has an additional field for the query and response code

Some errors in the query flow may not return the enhanced error
An example would be, the query returns valid data but the json.Unmarshal function in api.go returns an error.
This error is a default error which would not contain the query or status code, nor would it make sense to. the provider would still receive the query results, but the error would be related to the failure of the function Unmarshal.

Another example is if the query fails to build. This returns a generic error to the provider, which does not have the other fields. 

For these reasons, I am specifically checking whether the error received is a custom error of type client.Error, and handling those errors separately.